opentech/apply/funds/models.py

```diff
@@ -55,16 +61,31 @@ def process_form_submission(self, form):
                 response = cleaned_data.pop(field.id)
                 cleaned_data[field.block.name] = response
 
+        if form.user.is_authenticated():
+            user = form.user
+            cleaned_data['email'] = user.email
+            cleaned_data['full_name'] = user.get_full_name()
+        else:
+            User = get_user_model()
+            email = cleaned_data.get('email')
+            full_name = cleaned_data.get('full_name')
+            user = User.get_or_create_new(
```

Comment:
Is it worth adding a check that the user model has a `get_or_create_new` method? The test will fail if the user model is changed at a later point, so that's a decent safeguard. But something tells me ought to be defensive